amivapi/tests/events/test_emails.py

```diff
@@ -229,3 +229,39 @@ def test_no_nones_in_emails(self):
         mail = self.app.test_mails[1]
         for field in mail.values():
             self.assertTrue('None' not in field)
+
+    def test_signup_added_by_admin_accepted_emails(self):
+        """Test that an accepted email is sent when an admin has added a
+        signup manually."""
+        event = self.new_object('events', spots=1, selection_strategy='fcfs',
+                                allow_email_signup=True)
+
+        user = self.new_object('users')
+
+        self.api.post('/eventsignups', data={
+            'user': str(user['_id']),
+            'event': str(event['_id']),
+            'accepted': True
+        }, token=self.get_root_token(), status_code=201).json
+
+        mail = self.app.test_mails[0]
+        for field in mail.values():
+            self.assertTrue('None' not in field)
```

Comment:
Does it have to be `'None'` and not just `None`? (Not sure, I am not familiar with the mail tests, sorry).

Reply:
We have to check for the string `'None'` to make suer that neither the subject nor the email body contain that phrase as it would indicate that some placeholder was not set properly.

Reply:
And subjet and body are guaranteed to be strings, they can never be none?

Reply:
I did not write the mail test so I am not familiar with this. Could you double check this please?

In particular, can you check why it's `'None'` (string) not not just `None`? If we know the reason I am happy to acccapt it, but like this its a bit shady.

Reply:
Well, the first test case with the None values in the templated string fields is pretty obvious to me. Nevertheless I just added a comment there which additionally explains it in English.

So to explain it again with an example:
A confirmation email for the event signup of event XYZ should be sent. This is a prepared message containing placeholders. So the event name is a placeholder which gets replaced by 'XYZ'. In case the event name is None for some weird reason, the event name then listed in the message is 'None'.

Reply:
this looks like a satisfactory explanation to me.